pkg/gitutil/git.go

```diff
@@ -47,7 +47,15 @@ func IsGitCloned(gitPath string) (bool, error) {
 
 // update will fetch origin and set HEAD to origin/HEAD.
 func update(destinationPath string) error {
-	return exec(destinationPath, "pull", "--ff-only", "-v")
+	if err := exec(destinationPath, "fetch", "-v"); err != nil {
+		return errors.Wrapf(err, "fetch index at %q failed", destinationPath)
+	}
+
+	if err := exec(destinationPath, "reset", "--hard", "@{upstream}"); err != nil {
+		return errors.Wrapf(err, "reset index at %q failed", destinationPath)
```

Comment:
you actually don't need to do err!=nil check for the last err, you can just return it (as errors.Wrapf will return nil when err==nil.

Reply:
cool, I didn't know that :sunglasses: